apps/remote_control/lib/lexical/remote_control/search/indexer/entry.ex

```diff
@@ -20,40 +21,71 @@ defmodule Lexical.RemoteControl.Search.Indexer.Entry do
   @type t :: %__MODULE__{
           application: module(),
           subject: subject(),
-          parent: entry_id(),
+          block_id: block_id(),
           path: Path.t(),
           range: Lexical.Document.Range.t(),
           subtype: entry_subtype(),
           type: entry_type(),
           updated_at: :calendar.datetime()
         }
 
+  alias Lexical.Identifier
+  alias Lexical.RemoteControl.Search.Indexer.Source.Block
   alias Lexical.StructAccess
 
   use StructAccess
 
-  def reference(path, id, parent, subject, type, range, application) do
-    new(path, id, parent, subject, type, :reference, range, application)
+  defguard is_structure(entry) when entry.type == :metadata and entry.subtype == :block_structure
+  defguard is_block(entry) when entry.id == entry.block_id
+
+  def block_structure(path, structure) do
+    %__MODULE__{
+      path: path,
+      subject: structure,
+      type: :metadata,
+      subtype: :block_structure,
+      updated_at: timestamp()
+    }
+  end
+
+  def reference(path, %Block{} = block, subject, type, range, application) do
+    new(path, Identifier.next_global!(), block.id, subject, type, :reference, range, application)
   end
 
-  def definition(path, id, parent, subject, type, range, application) do
-    new(path, id, parent, subject, type, :definition, range, application)
+  def block_definition(path, %Block{} = block, subject, type, range, application) do
+    definition(
+      path,
+      block.id,
+      block.parent_id,
+      subject,
+      type,
+      range,
+      application
+    )
   end
 
-  defp new(path, id, parent, subject, type, subtype, range, application) do
+  defp definition(path, id, block_id, subject, type, range, application) do
```

Comment:
I'm not quite understanding the difference between "block_definition" and "definition."

Reply:
One is the definition of something that starts a block, a function, defmodule, fn, def, etc. the other is a definition of a thing that doesn't define a block, like a variable or a module attribute.